R/print.data.table.R

```diff
@@ -88,6 +90,17 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
     rownames(toprint)[1L] = ""
   }
   if (quote) colnames(toprint) <- paste0('"', old <- colnames(toprint), '"')
+  if (trunc.cols) {
+    # allow truncation of columns to print only what will fit in console #XXXX
+    widths = dt_width(x, class, row.names, names(x))
```

Comment:
I'm now avoiding the coercion. I'm using the `toprint` object (so it is usually going to be quite small). Because `nchar()` works differently on matrices, I've adjusted the `nchar_width()` function, shown below.

```
nchar_width = function(x) {
  each_width = nchar(x, type = "width")
  widths = vector("numeric", ncol(x))
  for (i in 1L:ncol(x)){
    widths[i] = max(each_width[, i])
  }
  widths
}
```

Is it a problem to use a loop here? I figured the number of columns would likely be reasonable for the loop not to be excessive.

In what situation is `toprint` a `data.table` instead of a matrix? Treating it like a matrix is working for all of the tests I've done (large and small tables, `class=TRUE`, `col.names='none'`, etc.) and it is working consistently.